src/Http/BaseApplication.php

```diff
@@ -69,7 +70,11 @@ public function bootstrap()
      */
     public function routes($routes)
     {
-        require $this->configDir . '/routes.php';
+        if (!Router::$initialized) {
+            require $this->configDir . '/routes.php';
+            // Prevent routes from being loaded again
+            Router::$initialized = true;
```

Comment:
Won't this mean that user overrides will need to set this flag as well?

Reply:
If they load the routes file then probably yes. If they load their own routes then it shouldn’t be necessary. 

Shouldn’t the flag be deprecated as well since there will be no `routes.php` in 4.0?

Reply:
It should. The `Router::_loadRoutes()` is already deprecated.